src/analysis/mean_inequalities.lean

```diff
@@ -600,6 +600,85 @@ begin
   { apply finset.sum_congr rfl (λ i hi, _), simp [H'.1 i hi, H'.2 i hi] }
 end
 
+lemma add_rpow_le_one_of_add_le_one {p : ℝ} (a b : ennreal) (hab : a + b ≤ 1) (hp1 : 1 ≤ p) :
+  a ^ p + b ^ p ≤ 1 :=
+begin
+  have h_le_one : ∀ x : ennreal, x ≤ 1 → x ^ p ≤ x, from λ x hx, rpow_le_self_of_le_one hx hp1,
+  have ha : a ≤ 1, from (self_le_add_right a b).trans hab,
+  have hb : b ≤ 1, from (self_le_add_left b a).trans hab,
+  exact (add_le_add (h_le_one a ha) (h_le_one b hb)).trans hab,
+end
+
+lemma add_rpow_le_rpow_add {p : ℝ} (a b : ennreal) (hp1 : 1 ≤ p) :
+  a ^ p + b ^ p ≤ (a + b) ^ p :=
+begin
+  by_cases h_top : a + b = ⊤,
+  { rw ←@ennreal.rpow_eq_top_iff_of_pos (a + b) p (lt_of_lt_of_le zero_lt_one hp1) at h_top,
+    rw h_top,
+    exact le_top, },
+  have ha_top : a ≠ ⊤,
+  { intro ha,
+    rw [ha, top_add] at h_top,
+    exact h_top rfl, },
```

Comment:
Should we extract a lemma saying that `a + b ≠ ⊤` implies `a ≠ ⊤`?

Reply:
it appears that there is already ` theorem ennreal.add_ne_top {a b : ennreal} : a + b ≠ ⊤ ↔ a ≠ ⊤ ∧ b ≠ ⊤` and I should have used it. I'll change that.

Reply:
Great!  Maybe it would be useful to extract a similar lemma about zero for `nnreal` and `ennreal`:
```
a + b ≠ 0 ↔ a ≠ 0 ∧ b ≠ 0
```
You're effectively using that later in the proof.

Reply:
something about `a + b ≠ \bot` in a canonically_ordered_monoid maybe.

Reply:
it was almost already there, with `add_eq_zero_iff`